src/test/java/uk/gov/hmcts/reform/pip/account/management/service/SensitivityServiceTest.java

```diff
@@ -94,4 +122,34 @@ void checkClassifiedReturnsFalseWhenVerifiedButNotMatchingProvenance() {
             "Returned true for classified sensitivity when verified with incorrect provenance");
     }
 
+    @ParameterizedTest
+    @MethodSource("parameters")
+    void checkClassifiedReturnsTrueForAllowedThirdPartyRolesOnly(ListType listType, Roles roles,
+                                                                 boolean isAuthorised) {
+        PiUser piUser = new PiUser();
+        piUser.setRoles(roles);
+        piUser.setUserProvenance(UserProvenances.THIRD_PARTY);
+
+        assertEquals(isAuthorised, sensitivityService.checkAuthorisation(piUser, listType, Sensitivity.CLASSIFIED),
+                     String.format("Should return %s for list type %s and role %s", isAuthorised, listType, roles));
+    }
+
+    private static Stream<Arguments> parameters() {
+        return Stream.of(
```

Comment:
Another little nugget of greatness for me to steal :D

Reply:
haha u need to give me some of your python expertise for exchange :)